cpp/ql/test/README.md

```diff
@@ -0,0 +1,33 @@
+# C/C++ CodeQL tests
+
+This document provides additional information about the C/C++ CodeQL Tests located in `cpp/ql/test`.  See [Contributing to CodeQL](CONTRIBUTING.md) for general information about contributing to this repository.
+
+The tests can be run through Visual Studio Code.  Advanced users may also use the `odasa qltest` command.
+
+## Contributing to the tests
+
+We are keen to have unit tests for all of our QL code.
+
+Every query in `cpp/ql/src` (outside of `cpp/ql/src/experimental`) should have a test in the corresponding subdirectory of `cpp/ql/test/query-tests`. At a minimum, each query test shall contain one case that should be detected by the query, and one related case that should not.
+
+TODO: example?
+
+Library features should also have test coverage, in `cpp/ql/test/library-tests`.
```

Comment:
Can you maybe expand this a bit to include to which standard we test?

Reply:
As far as I'm aware we've never really agreed a policy for how to test library features, apart from that we do tend to write tests for every new feature.  We should discuss this.

How about: "Every public class (and predicate???) in `cpp/ql/src` should have at least one test, including a test that would fail if the class were empty"?

Reply:
Yes, that's definitely ambiguous...

Reply:
Fixed the ambiguity.